.github/workflows/publish_pypi.yml

```diff
@@ -1,38 +1,63 @@
 name: Build and publish package to PyPI
 
 on:
-  push:
-    tags:
-    - '*'
+  release:
+    types: [published]
   workflow_dispatch:
     inputs:
       target:
-        description: 'Deployment target. Can be "pypi" or "testpypi"'
-        default: "pypi"
+        description: 'Deployment target. Can be "pypi" or "testpypi", or blank to skip publishing.'
+        default: ""
 
 jobs:
-  publish_pypi:
-    name: Build wheels on ubuntu-latest
+  build:
+    name: Build sdist and wheel
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-python@v4
-        with:
-          python-version: 3.8
-      - name: Install dependencies
-        run: |
-          python -m pip install --upgrade pip
-          pip install build
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
       - name: Build package
-        run: python -m build
-      - name: Publish on PyPI
-        if: github.event.inputs.target == 'pypi'
+        run: pipx run build
```

Comment:
Should use something like [hynek/build-and-inspect-python-package](https://github.com/hynek/build-and-inspect-python-package) to build + check the products (uploads them as an artifact too).

Reply:
I've used it extensively but was thinking that we avoid it here – let's go ahead with it, then!

Reply:
I've decided not to use it for the reasons described above. Let me know if this doesn't work and we can discuss further :)